src/index.ts

```diff
@@ -25,50 +28,37 @@ const excludedFiles = ['__tests__', 'utils'];
 const rules = readdirSync(rulesDir)
   .map(rule => parse(rule).name)
   .filter(rule => !excludedFiles.includes(rule))
-  .reduce(
+  .reduce<Record<string, RuleModule>>(
     (acc, curr) =>
-      Object.assign(acc, { [curr]: importDefault(join(rulesDir, curr)) }),
+      Object.assign(acc, {
+        [curr]: importDefault(join(rulesDir, curr)) as RuleModule,
+      }),
     {},
   );
 
+const recommendedRules = Object.entries(rules).reduce(
+  (acc, [name, rule]) =>
+    rule.meta.docs?.recommended
```

Comment:
`docs` shouldn't be optional, can you narrow the type?

Reply:
Yeah its a bit annoying - the types provided by `@typescript-eslint` are almost perfect, except for these edge cases which are somewhat tedious to improve on at the source.

I can actually improve this typing without too much work, which is nice.